cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1760,11 +1760,15 @@ private boolean isPodToRestart(StatefulSet ss, Pod pod, boolean isAncillaryCmCha
                 isPodCaCertUpToDate &= isPodCaCertUpToDate(pod, ca);
             }
 
-            boolean isPodToRestart = !isPodUpToDate || !isPodCaCertUpToDate || isAncillaryCmChange || isCaCertsChanged;
+            boolean isPodToRestart = !isPodUpToDate || isAncillaryCmChange;
+
             boolean isSatisfiedBy = true;
-            // it makes sense to check maintenance windows if pod restarting is needed
-            if (isPodToRestart) {
+            if (isCaCertsChanged || !isPodCaCertUpToDate) {
                 isSatisfiedBy = isMaintenanceTimeWindowsSatisfied(dateSupplier);
+                if (isSatisfiedBy) {
+                    // ca cert has changed and the window is opened
+                    isPodToRestart = true;
+                }
```

Comment:
I am not sure whether we need this.
```
else {
    isPodToRestart = false;
}
```

Reply:
I was just thinking about it ... but I guess this would block all other rolling updates if there is also CA rolling update needed, or? Would that make sense?

Reply:
At the end the code could be this one ...

```java
            boolean isPodToRestart = !isPodUpToDate || isAncillaryCmChange;

            boolean isSatisfiedBy = true;
            if (!isPodToRestart) {
                if (isCaCertsChanged || !isPodCaCertUpToDate) {
                    isSatisfiedBy = isMaintenanceTimeWindowsSatisfied(dateSupplier);
                    isPodToRestart |= isSatisfiedBy;
                }
            }
```
if  isPodTorestart is true ... not need for checking CA changes, maintenance etc etc ... we'll have our rolling because other reasons, because the user wants it (config change, logging change, ...)
if isPodToRestart is false ... we check the CA changes and the related maintenance window and do a logical OR with isPodToRestart which can remain false ... or becoming true if maintenance window is satisfied

Reply:
For having the right logging it could be this one ...

```java
               if (!reasons.isEmpty()) {
                    if (isPodToRestart) {
                        log.debug("{}: Rolling pod {} due to {}",
                                reconciliation, pod.getMetadata().getName(), reasons);
                    } else {
                        if (!isSatisfiedBy) {
                            log.debug("{}: Potential pod {} rolling due to {} but maintenance time windows not satisfied",
                                    reconciliation, pod.getMetadata().getName(), reasons);
                        }
                    }
                }
```